test/lifecycle_module_test.dart

```diff
@@ -174,218 +254,468 @@ void main() {
+        });
+      });
+    }
 
-    test('should do nothing when module is resumed if it was not suspended',
-        () async {
-      await module.load();
-      module.eventList.clear();
-      await module.resume();
-      expect(module.eventList, equals([]));
-    });
+    group('load', () {
+      var expectedLoadEvents = ['willLoad', 'onLoad', 'didLoad'];
+
+      test('should trigger loading events and call onLoad', () async {
+        await module.load();
+        expect(module.eventList, equals(expectedLoadEvents));
+      });
+
+      test('should set isLoading', () async {
+        expect(module.isLoading, isFalse);
+        var future = module.load();
+        expect(module.isLoading, isTrue);
+        await future;
+        expect(module.isLoading, isFalse);
+      });
+
+      test('should set isLoaded', () async {
+        expect(module.isLoaded, isFalse);
+        await module.load();
+        expect(module.isLoaded, isTrue);
+      });
+
+      test('should update module state', () async {
+        expectInLifecycleState(module, LifecycleState.instantiated);
+        var future = module.load();
+        expectInLifecycleState(module, LifecycleState.loading);
+        await future;
+        expectInLifecycleState(module, LifecycleState.loaded);
+      });
+
+      test('should return a useful future if it is already loading', () async {
+        await gotoState(module, LifecycleState.loading);
+        expect(module.isLoading, isTrue);
+        var future = module.load();
+        expect(module.isLoaded, isFalse);
+        await future;
+        expect(module.isLoaded, isTrue);
+      });
+
+      test('should return pending future if it is already loading', () async {
+        await gotoState(module, LifecycleState.loading);
+        var future1 = module.load();
+        var future2 = module.load();
+        expect(future1, same(future2));
+      });
+
+      test('should return new future if load is called after module has loaded',
+          () async {
+        await gotoState(module, LifecycleState.loading);
+        var future1 = module.load();
+        await future1;
+        var future2 = module.load();
+
+        expect(future1, isNot(same(future2)));
+      });
+
+      test('should only load once if it is already loading', () async {
+        var future = module.load();
+        expect(module.isLoaded, isFalse);
+        await Future.wait([module.load(), future]);
+        expect(module.isLoaded, isTrue);
+        expect(module.eventList, equals(expectedLoadEvents));
+      });
+
+      test('should warn if it is already loading', () async {
+        await gotoState(module, LifecycleState.loading);
+        expect(lastLogMessage, isNull);
+        await module.load();
+        expect(lastLogMessage, isNotNull);
+        expect(lastLogMessage.level, equals(Level.WARNING));
+      });
+
+      test('should do nothing if it was already loaded', () async {
```

Comment:
I don't think this test covers anything that the "should only load once if it is already loading" test doesn't.